src/api/index.ts

```diff
@@ -0,0 +1,48 @@
+export interface IUserCredentials {
```

Comment:
There should be no reference to `fetch` or `response.json()` in the component. All of this can easily be abstracted to functions in a module like this. It can also be broken down into different files for larger projects.

I have also used composition to create a more specific API. Notice that `request` is a general wrapper around `fetch`. Then there is an even more specific wrapper around `request` for `get` and `post`. When calling `post` we can simply provide the `body`.